pkg/util/azureclient/mgmt/resources/deployments_addons.go

```diff
@@ -22,8 +26,15 @@ func (c *deploymentsClient) CreateOrUpdateAndWait(ctx context.Context, resourceG
 	if err != nil {
 		return err
 	}
-
-	return future.WaitForCompletionRef(ctx, c.Client)
+	// all error codes: https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/common-deployment-errors
+	err = future.WaitForCompletionRef(ctx, c.Client)
+	if err != nil {
+		if detailedErr, ok := err.(autorest.DetailedError); ok &&
+			detailedErr.StatusCode == api.CloudErrorCodeResourceQuotaExceeded {
+			return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeResourceQuotaExceeded, "", detailedErr.Message, nil)
+		}
+	}
```

Comment:
_addons is the wrong place for this, it needs to be checked on the caller side.  Defining a helper somewhere `isResourceQuotaExceeded(error) bool` for DRY would be a good plan.